Caused by: java.io.IOException: Task did not succeed and did not report an error

version 3.0.10
I'm getting this error:

Caused by: java.io.IOException: Task did not succeed and did not report an error
	at com.sshtools.client.SshClient.doTask(SshClient.java:288)

This error message stinks. The reason it happens is because the throwable is getting saved here:
com.sshtools.common.ssh.ConnectionAwareTask#lastError
but the error is getting read from here:
com.sshtools.client.tasks.AbstractSessionTask#lastError

--- a/maverick-base/src/main/java/com/sshtools/common/ssh/ConnectionAwareTask.java
+++ b/maverick-base/src/main/java/com/sshtools/common/ssh/ConnectionAwareTask.java
@@ -25,7 +25,7 @@ import com.sshtools.common.logger.Log;
 public abstract class ConnectionAwareTask extends AbstractRequestFuture implements Runnable {
 
 	protected final SshConnection con;
-	private Throwable lastError;
+	protected Throwable lastError;
 	
 	public ConnectionAwareTask(SshConnection con) {
 		if(Objects.isNull(con)) {

--- a/maverick-synergy-client/src/main/java/com/sshtools/client/tasks/AbstractSessionTask.java
+++ b/maverick-synergy-client/src/main/java/com/sshtools/client/tasks/AbstractSessionTask.java
@@ -37,7 +37,6 @@ public abstract class AbstractSessionTask<T extends AbstractSessionChannel> exte
 	long timeout = 10000;
 	T session;
 	ChannelRequestFuture future;
-	Throwable lastError;
 	
 	public AbstractSessionTask(SshClient ssh, ChannelRequestFuture future) {
 		super(ssh);
@@ -59,10 +58,6 @@ public abstract class AbstractSessionTask<T extends AbstractSessionChannel> exte
 	
 	public void disconnect() {
 		con.disconnect();
-	}
-	
-	public final Throwable getLastError() {
-		return lastError;
 	}
 	
 	public ChannelRequestFuture getChannelFuture() {

